src/app/wallet/shared/address-table/address-table.component.html

```diff
@@ -30,16 +32,55 @@
         <span ngxClipboard [cbContent]="address.address" class="nc-icon nc-copy-2"></span>
         <!-- Open the modal for signing and verifying messages by address -->
         <span class="icon nc-icon nc-a-check"></span>
-        <span (click)="addressService.deleteAddress(address.address)" class="nc-icon nc-circle-remove color-red"></span>
+        <span (click)="deleteAddress(address.label, address.address)" class="nc-icon nc-circle-remove color-red"></span>
+
       </div>
     </div>
-    <div class="grid-cell divider text-center" *ngIf="displayQrMenu">
-      <div class="inline-toolbar qrcode-menu">
+    <div class="grid-cell divider text-center" *ngIf="!displayQrMenu">
+      <div class="inline-toolbar qrcode-menu" (click)="openQrCodeModal(address)">
         <span class="nc-icon nc-barcode-qr"></span>
       </div>
     </div>
   </div>
 </app-grid>
 
-<pagination *ngIf="displayPagination" [boundaryLinks]="true" (pageChanged)="pageChanged($event)" [totalItems]="addressService.addressCount" [maxSize]="5" [rotate]="false" [(ngModel)]="addressService.currentPage" class="pagination-sm center"
-            previousText="&lsaquo;" nextText="&rsaquo;" firstText="&laquo;" lastText="&raquo;" [itemsPerPage]="addressService.MAX_ADDRESSES_PER_PAGE" ></pagination>
+
+<pagination *ngIf="displayPagination" 
+  [boundaryLinks]="true" 
+  [totalItems]="getTotalAddressCount()" 
+  [maxSize]="5" 
+  [rotate]="false" 
+  [(ngModel)]="currentPage" 
+  class="pagination-sm center" previousText="&lsaquo;" nextText="&rsaquo;" firstText="&laquo;" lastText="&raquo;" 
+  [itemsPerPage]="getMaxAddressesPerPage()" >
+    
+  </pagination>
+
+<div id="qr-modal"  [ngClass]="{'hide': openQrModal == false }">
```

Comment:
We should only output if it is for the address-book page and not if it is for the receive page..
Maybe just add *ngIf="!displayQrMenu"?